src/Utils/Image.php

```diff
@@ -322,14 +322,14 @@ public function resize($width, $height, $flags = self::FIT)
 	public static function calculateSize($srcWidth, $srcHeight, $newWidth, $newHeight, $flags = self::FIT)
 	{
 		if (is_string($newWidth) && substr($newWidth, -1) === '%') {
-			$newWidth = (int) round($srcWidth / 100 * abs($newWidth));
+			$newWidth = (int) round($srcWidth / 100 * substr($newWidth, 0, -1));
```

Comment:
https://github.com/nette/utils/commit/8683a0bb07d80359b0226dd53f0257b734a74b9d was merged. ```abs``` is present in that commit.